### PR TITLE
Various metadata-related fixes and improvements.

### PR DESCRIPTION
**Pull request by tzakharko**
- If a value list variable has no values (all missing), the json value list metadata
  is now serialized as an empty list `[]` for consistency
- `NPStructurePresence` is no longer classified as a `PerLanguageSummaries` dataset
- `LID` field was sometimes serialized as string, fixed
- Missing glottocodes were sometimes serialized as explicit "NA" string, fixed
- Multiple metadata fixes:
  - Added value list descriptions for `PhonologicalFusion::FusionBinned6` and all variables that
    rely on it (such as `GrammaticalMarkers::MarkerFusionBinned6`)
  - Added value list descriptions for `PositionalBehavior::MarkerBehaviorBinned4` and all variables
    that rely on it (such as `GrammaticalMarkers::MarkerBehaviorBinned4`)
  - Value list description for `LocusOfMarking::LocusOfMarkingBinned5` was missing the value
    'FloatingorClitic', fixed (this also fixes all the variables that rely on it, such as
    `GrammaticalMarkers::LocusOfMarkingBinned5`)
  - Fixed value list description for `GrammaticalMarkers::MarkerPositionBinned4`
  - Fixed value list description for `GrammaticalMarkers::MarkerPositionBinned5`
  - Fixed data type of `GrammaticalMarkers::MarkerExpressesMultipleCategories` to be `logical`
  - Added value list descriptions for `ClauseLinkage::IntuitiveClassification`, value "?" is now
    recoded as NA (missing)
  - Added value list descriptions for multiple fields in `ClauseLinkage` where they were missing.
    The fields are: `AnticipatoryArgumentMarking`, `CataphoraConstraints`, `CategoricalSymmetry`,
    `ClauseLayer`, `ClausePosition`, `Embedding`, `ExtractionConstraints`, `FinitenessSimplified`,
    `FocusMarkingInDependent`, `FocusMarking`, `IllocutionaryMarking`, `IllocutionaryScope`,
    `InterpropositionalSemanticRelation`, `ReferenceTrackingSystem`, `TenseMarking` and
    `TenseScope`
  - Fixed the value list description for `ClauseWordOrder::WordOrderAPLex`
  - Fixed the value list description for `SemanticClass::SemanticClassBinned`
  - Removed invalid values from `GrammaticalRelationsRaw::SelectedArguments::SemanticCondition`
  - Fixed the value list descriptionb for `Register::OriginContinent`
  - Computed variables in `GrammaticalMarkersPerLanguage` now have correct value list metadata
  - Computed variables in `LocusOfMarkingPerLanguage` now have correct value list metadata
  - Computed variables `MorphologyPerLanguage::HasAny*` are now correctly annotated as logical
  - Computed variables `NPStructurePerLanguage::NPHas*` are now correctly annotated as logical
  - `NPStructurePerLanguage::NPStructureID` is now correctly annotated as integer
  - Computed variables in `VerbInflection*` summary datasets now have correct value list metadata

### CODE REVIEW

File: metadata/PerLanguageSummaries/VerbInflectionMacrocategoriesAggregatedByMarkerPosition.yaml [hunk 1]
```diff
@@ -41,82 +41,207 @@ fields:
...
+        after the phonological host
   # -- end of MarkerPositionForNPRelated
 
   MarkerPositionForClassification:
     description: >-
       GrammaticalMarkers::MarkerPosition value for Classification
     kind: computed data (aggregation-scripts/VerbInflectionPerLanguage.R)
-    data: logical
+    data: value-list
+    values: []
```
Comment (xrotwang) [inline metadata/PerLanguageSummaries/VerbInflectionMacrocategoriesAggregatedByMarkerPosition.yaml]: This is supposed to be a dictionary, not a list.

Comment (xrotwang) [reply metadata/PerLanguageSummaries/VerbInflectionMacrocategoriesAggregatedByMarkerPosition.yaml]: There's a couple more like this.

Comment (tzakharko) [reply metadata/PerLanguageSummaries/VerbInflectionMacrocategoriesAggregatedByMarkerPosition.yaml]: Good point. Fixed in c52d68af4646034adc58e32875f62d1669bbcf9c

**Review by xrotwang — COMMENTED** [review]
Ahom has a glottocode: https://glottolog.org/resource/languoid/id/ahom1240

**Comment by tzakharko** [comment]
Updated the glottocode for Ahom and  Eastern Armenian in the database, will be reflected on the next data export. The only language with a missing glottocode is now Serbian Torlak, which I could not locate in Glottolog (there is a note mentioning this variety under standard Serbian though)

**Comment by xrotwang** [comment]
@tzakharko see https://github.com/glottolog/glottolog/issues/815#issuecomment-1048583147

**Comment by tzakharko** [comment]
> @tzakharko see [glottolog/glottolog#815 (comment)](https://github.com/glottolog/glottolog/issues/815#issuecomment-1048583147)

Thanks, I forwarded this to the team, will update the glottocode once a decision is made.

**Comment by tzakharko** [comment]
@xrotwang All the issues you have found so far should be fixed now. If you don't see any other low-hanging issues, I would like to publish a bugfix release based on the contents of this branch.

**Comment by xrotwang** [comment]
It would be nice, if the issue(s) with the bibliography could be addressed, too. That would save me a couple of lines cleaning it up.
Otherwise, yes, all issues addressed.

**Comment by tzakharko** [comment]
> It would be nice, if the issue(s) with the bibliography could be addressed, too. That would save me a couple of lines cleaning it up.

Yes, of course! Adopted your version of the bibliography in latest commit.

**Comment by xrotwang** [comment]
So the bibliography isn't pulled out of AUTOTYP but only exists as this BibTeX file anyway?

**Comment by tzakharko** [comment]
It is pulled from the database, but that particular part of the database is currently on life support and is not actively updated.  This is another part that will require an overhaul, as some sources are "hidden" in the comments or special fields of various data file and are not currently exported. It is a lot of legacy to deal with.. 

For now your curated file works great, and as we continue modernising the database structure we will put a new bibliogaphy-generating mechanism in place. Making use of the excellent Glottolog reference database sounds like an obvious choice here.

**Pull request closed without merging**